### PR TITLE
Add public total count measure counting the total number of results f…

### PR DESCRIPTION
**Pull request by mdellabani**
Implements #167:

- Add a `total_count` measure counting the total number of results not considering the query limit
- Add a totalCount field in the `QueryResultDto`

### CODE REVIEW

File: js/typescript-library/src/generate-json-queryDto.ts [hunk 1]
```diff
@@ -56,6 +57,7 @@ export function generateFromQueryDto() {
   const expression = new ExpressionMeasure("myExpression", "sum(price*quantity)")
   q.withMeasure(expression)
   q.withMeasure(count)
+  q.withMeasure(total_count)
```
Comment (mdellabani) [inline js/typescript-library/src/generate-json-queryDto.ts:60]: modify java test

File: server/src/test/java/io/squashql/client/HttpClientQuerierTest.java [hunk 2]
```diff
@@ -137,7 +139,7 @@ void testRunQueryWithCondition() {
     // Note. The CJ will make null appear in rows. We want to make sure null values are correctly handled.
     QueryDto query = Query
             .from("our_prices")
-            .where(SCENARIO_FIELD_NAME, Functions.eq(MAIN_SCENARIO_NAME))
```
Comment (paulbares) [inline server/src/test/java/io/squashql/client/HttpClientQuerierTest.java:140]: Could you revert this condition, test is not passing.

File: js/typescript-library/src/measure.ts [hunk 3]
```diff
@@ -57,6 +57,8 @@ export class ExpressionMeasure implements BasicMeasure {
   }
 }
 
+export const total_count = new ExpressionMeasure("_total_count_", "COUNT(*) OVER ()")
```
Comment (paulbares) [inline js/typescript-library/src/measure.ts:60]: total_count could you use camelCase style: `totalCount` ?

**Review by paulbares — CHANGES_REQUESTED** [review]
2 small changes to do

**Review by paulbares — CHANGES_REQUESTED** [review]
One more thing, could you add a test in `ATestQueryExecutor` with a limit set to 1 and check the correct totalCount value is returned. Tests in this class will be run with all compatible databases to make sure the measure is working.

File: server/src/test/java/io/squashql/client/HttpClientQuerierTest.java (hunk 2 above)
Comment (mdellabani) [reply server/src/test/java/io/squashql/client/HttpClientQuerierTest.java:140]: My bad typo when was editing

**Review by paulbares — APPROVED** [review]

**Pull request merged**